src/sentry/integrations/request_buffer.py

```diff
@@ -0,0 +1,143 @@
+import logging
+from datetime import datetime, timedelta
+
+from django.conf import settings
+
+from sentry.utils import json, redis
+
+BUFFER_SIZE = 30  # 30 days
+KEY_EXPIRY = 60 * 60 * 24 * 30  # 30 days
+
+IS_BROKEN_RANGE = 7  # 7 days
+
+
+class IntegrationRequestBuffer:
+    """
+    Create a data structure to store daily successful and failed request counts for each installed integration in Redis
+    This should store the aggregate counts of each type for last 30 days for each integration
+    """
+
+    def __init__(self, key):
+        self.integrationkey = key
+
+        logger = logging.getLogger(__name__)
+
+        try:
+            cluster_id = settings.SENTRY_INTEGRATION_ERROR_LOG_REDIS_CLUSTER
+            self.client = redis.redis_clusters.get(cluster_id)
+        except KeyError as e:
+            logger.info("no_redis_cluster", extra={"error": e, "cluster_id": cluster_id})
+
+    def _convert_obj_to_dict(self, redis_object):
+        """
+        Convert the request string stored in Redis to a python dict
+        """
+
+        return json.loads(redis_object)
+
+    def _get_all_from_buffer(self, buffer_key):
+        """
+        Get the list at the buffer key.
+        """
+
+        # return self.client.lrange(buffer_key, 0, BUFFER_SIZE - 1)
+        return sum(
+            [
+                self.client.lrange(key, 0, BUFFER_SIZE - 1)
+                for key in self.client.keys(buffer_key + "*")
```

Comment:
The keys command is not intended for production use, see the warning on it's [command page](https://redis.io/commands/keys/). Also note that although they mention scan as an alternative, it is also not really appropriate for production use.